sdk/core/perf/src/base_test.cpp

```diff
@@ -0,0 +1,197 @@
+
+    // Copy all headers
+    for (auto& header : request.GetHeaders())
+    {
+      redirectRequest.SetHeader(header.first, header.second);
+    }
+    // QP
+    for (auto const& qp : request.GetUrl().GetQueryParameters())
+    {
+      redirectRequest.GetUrl().AppendQueryParameter(qp.first, qp.second);
+    }
+    // Set x-recording-upstream-base-uri
+    {
+      auto const& url = request.GetUrl();
+      auto const port = url.GetPort();
+      auto const host
+          = url.GetScheme() + "://" + url.GetHost() + (port != 0 ? ":" + std::to_string(port) : "");
+      redirectRequest.SetHeader("x-recording-upstream-base-uri", host);
+    }
+    // Set recording-id
+    redirectRequest.SetHeader("x-recording-id", RecordId);
+    redirectRequest.SetHeader("x-recording-remove", "false");
+
+    // Using recordId, find out MODE
+    if (IsPlayBackMode)
+    {
+      // PLAYBACK mode
+      redirectRequest.SetHeader("x-recording-mode", "playback");
+    }
+    else
+    {
+      // RECORDING mode
+      redirectRequest.SetHeader("x-recording-mode", "record");
+    }
+
+    return nextPolicy.Send(redirectRequest, context);
+  }
+
+  std::unique_ptr<HttpPolicy> Clone() const override
+  {
+    return std::make_unique<ProxyPolicy>(*this);
+  }
+};
+
+} // namespace
+
+namespace Azure { namespace Perf {
+
+  void BaseTest::ConfigureCoreClientOptions(Azure::Core::_internal::ClientOptions* clientOptions)
+  {
+    std::string proxy(m_options.GetOptionOrDefault("Proxy", ""));
+    if (!proxy.empty())
+    {
+      // If proxy is set in the options, the proxy policy is attached in RECORD MODE
+      clientOptions->PerRetryPolicies.push_back(std::make_unique<ProxyPolicy>(proxy));
+    }
+  }
+
+  void BaseTest::PostSetUp()
+  {
+    std::string proxy(m_options.GetOptionOrDefault("Proxy", ""));
+
+    if (!proxy.empty())
+    {
+      Azure::Core::_internal::ClientOptions clientOp;
+      clientOp.Retry.MaxRetries = 0;
+      std::vector<std::unique_ptr<Azure::Core::Http::Policies::HttpPolicy>> policiesOp;
+      std::vector<std::unique_ptr<Azure::Core::Http::Policies::HttpPolicy>> policiesRe;
+      Azure::Core::Http::_internal::HttpPipeline pipeline(
+          clientOp, "PerfFw", "na", std::move(policiesRe), std::move(policiesOp));
+      Azure::Core::Context ctx;
+
+      // Send start-record call
+      {
+        Azure::Core::Url startRecordReq(proxy);
+        startRecordReq.AppendPath("record");
+        startRecordReq.AppendPath("start");
+        Azure::Core::Http::Request request(Azure::Core::Http::HttpMethod::Post, startRecordReq);
+        auto response = pipeline.Send(request, ctx);
+
+        auto const& headers = response->GetHeaders();
+        auto findHeader = std::find_if(
+            headers.begin(),
+            headers.end(),
+            [](std::pair<std::string const&, std::string const&> h) {
+              return h.first == "x-recording-id";
+            });
+        RecordId.append(findHeader->second);
```

Comment:
Is this appending to an existing string?  Would assignment be clearer/safer?  I don't think we ever want to **append** a recording ID to an existing string.

Reply:
I was using a global string `recordId` and it was easier clear and append because the recordId is always the same size. Which means the string content can re-use the same heap memory all the time.  But it was a tiny insignificant optimization.

I changed it to support multiple proxies anyway. The recordId is now inside each test